Makefile

```diff
@@ -1,28 +1,31 @@
 ###### Help ###################################################################
-
 .DEFAULT_GOAL = help
 
 .PHONY: help
 help: ## list Makefile targets
 	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
 
-###### Targets ################################################################
-
+###### Setup ##################################################################
 IAAS=azure
+CSB_VERSION := $(or $(CSB_VERSION), $(shell grep 'github.com/cloudfoundry/cloud-service-broker' go.mod | grep -v replace | awk '{print $$NF}' | sed -e 's/v//'))
+CSB := $(or $(CSB), cfplatformeng/csb:$(CSB_VERSION))
+GO_OK := $(shell which go 1>/dev/null 2>/dev/null; echo $$?)
+DOCKER_OK := $(shell which docker 1>/dev/null 2>/dev/null; echo $$?)
+ifeq ($(GO_OK), 0)
+GO=go
+BUILDER=go run github.com/cloudfoundry/cloud-service-broker
+LDFLAGS="-X github.com/cloudfoundry/cloud-service-broker/utils.Version=$(CSB_VERSION)"
+GET_CSB="env CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags $(LDFLAGS) github.com/cloudfoundry/cloud-service-broker"
+else ifeq ($(DOCKER_OK), 0)
 DOCKER_OPTS=--rm -v $(PWD):/brokerpak -w /brokerpak --network=host
-CSB := $(or $(CSB), cfplatformeng/csb)
-USE_GO_CONTAINERS := $(or $(USE_GO_CONTAINERS), 1)
-
-ifeq ($(USE_GO_CONTAINERS), 0)
-BUILDER=./cloud-service-broker
-else
+GO=docker run $(DOCKER_OPTS) golang:$(GOVERSION) go
 BUILDER=docker run $(DOCKER_OPTS) $(CSB)
+GET_CSB="wget -O cloud-service-broker https://github.com/cloudfoundry/cloud-service-broker/releases/download/v$(CSB_VERSION)/cloud-service-broker.linux && chmod +x cloud-service-broker"
+else
+$(error either Go or Docker must be installed)
 endif
 
-.PHONY: acceptance-tests
```

Comment:
Might be good to look into something like [include](https://www.gnu.org/software/make/manual/make.html#Include), as this bit is getting repeated across Makefiles